File: Atcoder/Ruby/145.ABC/B.rb
kes = gets.to_i
fuji = gets.chomp

mori = fuji.size
yama = fuji[0..mori/2-1]
moriyama = yama*2


if(kes%2 != 0)
  puts 'No'
  exit
elsif(moriyama != fuji)
  puts 'No'
  exit
end
puts 'Yes'

File: Atcoder/PHP/145.ABC/B.php
<?php
fscanf(STDIN, "%d", $kes);
$mori = trim(fgets(STDIN));
$mori1 = str_split($mori);
$yama = array_slice($mori1, 0, $kes/2);
$yama1 = implode("", $yama);
$taku = $yama1.$yama1;

if($kes%2 != 0){
  echo 'No';
}elseif($kes%2 == 0 && $mori != $taku){
  echo 'No';
}else{
  echo 'Yes';
}